src/test/java/org/jenkinsci/plugins/credentialsbinding/masking/BatchPatternMaskerProviderTest.java

```diff
@@ -0,0 +1,367 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.credentialsbinding.masking;
+
+import hudson.Functions;
+import hudson.model.Result;
+import org.jenkinsci.plugins.credentialsbinding.test.CredentialsTestUtil;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.IOException;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assume.assumeTrue;
+
+public class BatchPatternMaskerProviderTest {
+
+    private static final String SIMPLE = "abcABC123";
+    private static final String SAMPLE_PASSWORD = "}#T14'GAz&H!{$U_";
+    private static final String ESCAPE = "^<^>^(^)^^^&^|";
+
+    // ALL_ASCII - [ < > " ' ^ $ | % ] 
+    private static final String NON_DANGEROUS = "!#$*+,-./ 0123456789:;=? @ABCDEFGHIJKLMNO PQRSTUVWXYZ[\\]_ `abcdefghijklmno pqrstuvwxyz{}~";
+
+    // <>"'^$|
+    private static final String NON_DANGEROUS_IN_DOUBLE = "abc<def>$ghi|jkl";
+
+    // quoted form: "^^|\a\\"
+    private static final String NEEDS_QUOTING = "^|\\a\\";
+
+    private static final String ALL_ASCII = "!\"#$%&'()*+,-./ 0123456789:;<=>? @ABCDEFGHIJKLMNO PQRSTUVWXYZ[\\]^_ `abcdefghijklmno pqrstuvwxyz{|}~";
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    private WorkflowJob project;
+
+    private String credentialPlainText;
+    private String credentialId;
+
+    @Before
+    public void assumeWindowsForBatch() throws Exception {
+        assumeTrue(Functions.isWindows());
+    }
+
+    private void registerCredentials(String password) throws IOException {
+        this.credentialPlainText = password;
+        this.credentialId = CredentialsTestUtil.registerStringCredentials(j.jenkins, password);
+    }
+
+    private WorkflowRun runProject() throws Exception {
+        return j.assertBuildStatusSuccess(project.scheduleBuild2(0));
+    }
+
+    @Test
+    public void simple_noQuote() throws Exception {
+        registerCredentials(SIMPLE);
+        assertDirectNoPlainTextButStars(runDirectNoQuote());
+    }
+
+    @Test
+    public void simple_singleQuote() throws Exception {
+        registerCredentials(SIMPLE);
```

Comment:
Why is the Bash version using the theory system, and this is done manually? Smells like there could be some consolidation of test code.

Reply:
Different test authors originally. I can likely figure out how to break this into a theorized test.